### DISCUSSION TITLE
Reduced Order Modelling using MFEM as a Full Order Model

### DISCUSSION THREAD

**Discussion by JacobLotz** [mfem/mfem#3368] [Q&A] [answered]
Hello MFEM users,

I am currently starting a project in which I want to use MFEM as a Full Order Model to create a Reduced Order Model. I found the [libROM code](https://www.librom.net/features.html) which seems to be very suitable to generate a Reduced Order Model using MFEM as a Full Order Model. As it has many MFEM examples in it, they should go well together. However, to broaden the horizon, does anyone have experience with / recommendations for other ROM libraries which work well with MFEM as a Full Order Model?

Kind regards,

Jacob Lotz

**Comment by tzanio** [comment]
ping: @dylan-copeland

**Comment by dylan-copeland** [comment, marked as answer]
Thanks @JacobLotz for the question. I am one of the main developers of the libROM library, as well as an MFEM developer, so I may have some bias. Of course, I strongly recommend libROM, as it has many features and the ability to support a wide range of applications with different types of ROM methods.

An important question in choosing a ROM library is what kind of ROM method you would like to use. ROM methods range in code-invasiveness, from the very invasive projection-based ROMs to non-invasive, purely data-driven methods such as DMD and latent space dynamics identification (all of these are supported in libROM). For the data-driven methods, other libraries may be easy to use with MFEM, since they do not involve the FOM code. On the other hand, for projection-based ROMs, there are some non-trivial things to implement in the FOM code, especially hyperreduction on sample meshes. As far as I know, libROM is the only library that supports hyperreduction in MFEM. Also, libROM is the only ROM library with an MFEM team member to help. So please feel free to ask questions in the libROM repo on github.

If you have success with other ROM libraries applied to MFEM codes, we would also be very interested to hear about that.

**Reply by JacobLotz** [reply]
Thank you @dylan-copeland for your extensive reply! I am not sure yet which method I would like to use, but it will likely be a projection based method. libROM looks like the best fit up to now, so that will probably be the library of choice.

The ROM community of MFEM is probably small and if anyone else has experience with any other ROM libraries with MFEM as FOM I am still very much interested!

Kind regards, Jacob
